### PR TITLE
[SPARK-41943][CORE] Use java api to create files and grant permissions

### PR DESCRIPTION
**Pull request by smallzhongfeng**
### What changes were proposed in this pull request?

For method `createDirWithPermission770`, using java api to create files and grant permissions instead of calling shell commands.

### Why are the changes needed?

Safer and more efficient.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Origin uts.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala [hunk 1]
```diff
@@ -301,9 +301,6 @@ private[spark] class DiskBlockManager(
    * Create a directory that is writable by the group.
    * Grant the permission 770 "rwxrwx---" to the directory so the shuffle server can
    * create subdirs/files within the merge folder.
-   * TODO: Find out why can't we create a dir using java api with permission 770
```
Comment (smallzhongfeng) [inline core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala:304]: For `TODO`, maybe we can see [https://bugs.openjdk.org/browse/JDK-8220013](https://bugs.openjdk.org/browse/JDK-8220013)

**Comment by mridulm** [comment]
+CC @zhouyejoe

**Comment by smallzhongfeng** [comment]
cc  @cloud-fan @HyukjinKwon @LuciferYang @@zhouyejoe Hope to get your opinion. :)

File: core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala (hunk 1 above)
Comment (HyukjinKwon) [reply core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala:304]: Mind summarizing why we can use this approach while the JDK issue isn't resolved?

Comment (smallzhongfeng) [reply core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala:304]: In unix like systems every process has a property called umask which is masked onto the permissions of any file created and inherited by child processes. the default is 0002 or "turn off write for others". So it's most likely that Java is setting the permission you seek and then it's being masked out. At present, this issue has not been resolved, so my current approach is to create a directory first, and then explicitly grant permissions.

**Review by cloud-fan — APPROVED** [review]

**Comment by zhouyejoe** [comment]
There was a long discussion thread regarding this implementation in this [PR](https://github.com/apache/spark/pull/35085#discussion_r786892744). There will be some issue with setgid. 
Will add @otterc @akpatnam25 @Kimahriman for additional comments and review.

**Comment by smallzhongfeng** [comment]
> There was a long discussion thread regarding this implementation in this [PR](https://github.com/apache/spark/pull/35085#discussion_r786892744). There will be some issue with setgid. Will add @otterc @akpatnam25 @Kimahriman for additional comments and review.

Thank you for reminding me @zhouyejoe . I looked at the results of the discussion and it seems to be consistent with my handling method. You can see https://github.com/apache/spark/pull/35085#discussion_r813523793

**Comment by Kimahriman** [comment]
Yeah I think this is fine. `mkdir -p` essentially does a `mkdir` syscall followed by a `chmod` sys call, which is effectively what this is doing as well. Both approaches lose the setgid bit which is why all the make shuffle files world readable was necessary in https://github.com/apache/spark/pull/35085

**Review by zhouyejoe — APPROVED** [review]

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Comment by smallzhongfeng** [comment]
Thanks all. @mridulm @zhouyejoe @Kimahriman @HyukjinKwon @cloud-fan

**Pull request closed without merging**